tensordict/prototype/tensorclass.py

```diff
@@ -63,7 +63,11 @@
 def is_tensorclass(obj: type | Any) -> bool:
     """Returns True if obj is either a tensorclass or an instance of a tensorclass."""
     cls = obj if isinstance(obj, type) else type(obj)
-    return dataclasses.is_dataclass(cls) and hasattr(cls, "__TENSORCLASS")
+    return (
+        dataclasses.is_dataclass(cls)
+        and "_non_tensordict" in cls.__dict__
+        and "_tensordict" in cls.__dict__
```

Comment:
`is_tensorclass` is giving the wrong answer here because `_non_tensordict` and `_tensordict` are instance attributes rather than class attributes.